src/index.ts

```diff
@@ -106,19 +107,25 @@ class MagicNixCacheAction extends DetSysAction {
   }
 
   async post(): Promise<void> {
-    if (this.noopMode) {
-      actionsCore.debug(TEXT_NOOP);
-      return;
-    }
+    // If strict mode is off and there was an error in main, such as the daemon not starting,
+    // then the post phase is skipped with a warning.
+    if (!this.strictMode && this.mainError) {
+      this.exitWithWarning(this.mainError);
+    } else {
+      if (this.noopMode) {
```

Comment:
what is noopMode? Should that conditional be pulled out and be a first fast exit? Or can mainError happen even in noopMode?

Reply:
Ah, you're right. No-op mode pretty much does what I'm trying to accomplished with the "error on main" logic. I'll remove it.

Reply:
Actually, strike that: the difference is that "no-op mode" means that the daemon is already running and that it doesn't need to be started again. So having both "branches" makes sense here. But maybe I can adjust the naming to make it more intuitive.